Ginger/GingerCoreNET/RunLib/CLILib/CLIProcessor.cs

```diff
@@ -40,26 +42,114 @@ public void ExecuteArgs(string[] args)
 
             Reporter.ToLog(eLogLevel.DEBUG, string.Format("########################## Starting Automatic {0} Execution Process ##########################", GingerDicser.GetTermResValue(eTermResKey.RunSet)));
 
-            Reporter.ToLog(eLogLevel.DEBUG, string.Format("Parsing {0} execution arguments...", GingerDicser.GetTermResValue(eTermResKey.RunSet)));                          
+            Reporter.ToLog(eLogLevel.DEBUG, string.Format("Parsing {0} execution arguments...", GingerDicser.GetTermResValue(eTermResKey.RunSet)));
             ConsoleWorkspaceEventHandler consoleWorkspaceEventHandler = new ConsoleWorkspaceEventHandler();
-            string param;
-            string value = null;
-            //if (args[0].StartsWith("ConfigFile=") || args[0].StartsWith("DynamicXML="))  // special case to support backword compatibility of old style ConfigFile=%filename%
-            if (args[0].Contains("="))
+
+            // This is the new style using verb and CommandLine Nuget to parse
+            if (args[0] == "run" || args[0] == "grid")
             {
-                string[] arg1 = args[0].Split(new[]{ '='}, 2);
-                param = arg1[0].Trim();
-                value = arg1[1].Trim();                
+                ParseArgs(args);
             }
             else
-            {                
-                param = args[0].Trim();
-                if (args.Length > 1)
+            {
+                // Support old style
+                string param;
+                string value = null;                
+                if (args[0].Contains("="))
+                {
+                    string[] arg1 = args[0].Split(new[] { '=' }, 2);
+                    param = arg1[0].Trim();
+                    value = arg1[1].Trim();
+                }
+                else
+                {
+                    param = args[0].Trim();
+                    if (args.Length > 1)
+                    {
+                        value = args[1].Trim();
+                    }
+                }
+                HandleArg(param, value);
+            }
+
+        }
+
+        private void ParseArgs(string[] args)
+        {            
```

Comment:
i think we should call this ParseArgs inside CLIArgs.LoadContent() in case we identify the the arguments are in new style or use existing if old because anyway it been used only for CLIArgs and not for others (as it should for my opinion)

Reply:
I want to change this Parse Args to do all args parsing old and new, if old then it will convert the args old style to new style and run the same engine, WDYT?

Reply:
for CLI with Arguments I think we have only few users started to use it so i'm ok with removing the support to old Args on 3.4 and support only new style. it will require Itsik approval